test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java

```diff
@@ -552,6 +583,127 @@ private EventWriterConfig buildWriterConfig() {
         return validEvents;
     }
 
+    void writeInKVP(String scope, String kvtName, String kf, int kvpCount, KeyValueTableFactory keyValueTableFactory) {
+        log.info("Start KVP write operation of KVT {} and scope {}", kvtName, scope);
+        // Creating 5 threads to update KVP table entry
+        List<CompletableFuture<Void>> writerList = new ArrayList<>();
+        for (Integer i = 1; i <= kvpCount; i++) {
+            String value = convertStringToBinary("Hello World");
+            Integer keyId = i;
+            writerList.add(writeKVP(scope, kvtName, kf, value, keyId, keyValueTableFactory));
+        }
+        Futures.allOf(writerList);
+        /*if (!Futures.await(Futures.allOf(writerList))) {
+            log.error("Writers stopped with exceptions");
+        }*/
+    }
+
+    private CompletableFuture<Void> writeKVP(String scope, String kvtName, String kf,
+                                     String value, Integer keyId, KeyValueTableFactory keyValueTableFactory) {
+        return CompletableFuture.runAsync(() -> {
```

Comment:
You are eating 2 threads to do this. One is for the `runAsync` and one is by the actual code. 

Rewrite this method to not use `runAsync` or `join`/`get` below; instead use CompletableFuture composition to write your code.

Reply:
Resolved, used supplyAsync and thenCompose